frontend/simple/views/components/ProposalBox/ProposalBox.vue

```diff
@@ -0,0 +1,305 @@
+            }"
+            :disabled="hasVoted"
+            @click="handleVoteFor"
+            data-test="forButton"
+          >
+            {{buttonText.for}}
+          </button>
+        </template>
+
+        <button-countdown
+          v-else-if="!isProposalClosed"
+          :onStateChange="handleProposalStateChange"
+        >
+          {{closeProposalBtnText}}
+        </button-countdown>
+      </div>
+
+      <p class="has-text-grey has-text-right"
+        :class="{ 'gi-feedback has-text-weight-bold': isProposalClosed }"
+      >
+        {{helperText}}
+      </p>
+    </div>
+  </div>
+</transition>
+</template>
+<style lang="scss" scoped>
+@import "../../../assets/sass/theme/index";
+
+%avatarSize {
+  width: 4.5rem;
+  height: 4.5rem;
+}
+
+.gi-voting {
+  flex-wrap: wrap;
+  justify-content: flex-end;
+  margin: $gi-spacer*3 0;
+
+  &:last-child {
+    margin-bottom: 0;
+  }
+
+  &-body {
+    margin-bottom: $gi-spacer;
+    flex-grow: 1;
+  }
+
+  &-info {
+    flex-grow: 1;
+    padding-left: $gi-spacer;
+
+    &-title {
+      margin-bottom: $gi-spacer-xs;
+      line-height: 1;
+    }
+  }
+
+  @include tablet {
+    flex-wrap: nowrap;
+    justify-content: space-between;
+
+    &-body {
+      margin-bottom: 0;
+    }
+
+    &-info {
+      padding: 0 $gi-spacer 0 $gi-spacer-lg;
+    }
+  }
+}
+
+.gi-feedback {
+  margin-top: $gi-spacer;
+}
+
+.buttons {
+  flex-wrap: nowrap;
+  justify-content: flex-end;
+}
+</style>
+<script>
+import ButtonCountdown, { countdownStates } from '../ButtonCountdown'
+import Sign from './Sign.vue'
+import L from '../../utils/translations'
+import template from 'string-template'
+import contracts from '../../../model/contracts.js'
+const { DoInvitation, DoRemoval, DoIncome, DoChangeThreshold, DoApprovalThreshold, DoRemovalThreshold } = contracts.GroupProposal
+
+export default {
+  name: 'ProposalBox',
+  props: {
+    proposal: {
+      type: [Object],
+      required: true
+    },
+    // Actions:
+    onVoteAgainst: Function,
+    onVoteFor: Function,
+    onCloseProposal: Function
+  },
+  components: {
+    ButtonCountdown,
+    Sign
+  },
+  data () {
+    return {
+      ephemeral: {
+        errorState: null,
+        closeCountdown: {
+          state: null,
+          countdown: null
+        }
+      }
+    }
+  },
+  computed: {
+    buttonText () {
+      const { doWhat, toWhat, fromWhat } = this.proposal
+      if (doWhat === DoInvitation) {
+        return {
+          for: template(L('Invite {toWhat}'), { toWhat }),
+          against: L('Don\'t invite')
+        }
+      } else if (doWhat === DoRemoval) {
+        return {
+          for: template(L('Remove {toWhat}'), { toWhat }),
+          against: template(L('Keep {toWhat}'), { toWhat })
+        }
+      } else {
+        return {
+          for: template(L('Change to {toWhat}'), { toWhat }),
+          against: template(L('Keep {fromWhat}'), { fromWhat })
+        }
+      }
+    },
```

Comment:
Instead of having this here, this should instead be a simple map defined in a different file and keyed by `proposalType`

Reply:
Or, it can be in a `const` map defined at the top of the file here